docs/dependency.md

```diff
@@ -1,3 +1,152 @@
+  },
+  assets: string[],
+  hooks: {
+    [key: string]: string,
+  }
+};
+```
+
+> Note: This interface is subject to breaking changes. We may consider renaming some keys to simplify the configuration further. If you are going to use it, be prepared to update before we ship a stable 0.60.0 release.
+
+### platforms
+
+A map of specific settings that can be set per platform. The exact shape is always defined by the package that provides given platform. 
+
+In most cases, as a library author, you should not need to define any of these.
+
+The following settings are available on iOS and Android:
+
+```ts
+type DependencyParamsIOST = {
+  project?: string,
+  sharedLibraries?: string[],
+};
+
+type DependencyParamsAndroidT = {
+  sourceDir?: string,
+  manifestPath?: string,
+  packageImportPath?: string,
+  packageInstance?: string
+};
+```
+
+#### platforms.ios.project
+
+Custom path to `.xcodeproj`
+
+#### platforms.ios.sharedLibraries
+
+An array of shared iOS libraries to link with the dependency. E.g. `libc++`. This is mostly a requirement of the native code that a dependency ships with.
+
+#### platforms.android.sourceDir
+
+Path to a folder with source files.
+
+#### platforms.android.manifestPath
+
+Path to a custom `AndroidManifest.xml`
+
+#### platforms.android.packageImportPath
+
+Custom package import. For example: `import com.acme.AwesomePackage;`.
+
+#### platforms.android.packageInstance
+
+Custom syntax to instantiate a package. By default, it's a `new AwesomePackage()`. It can be useful when your package requires additional arguments while initializing. 
+
+For settings applicable on other platforms, please consult their respective documentation.
+
+### assets
+
+An array of assets folders to glob for files to link.
+
+### hooks
+
+A map where key is the name of a hook and value is the path to a file to execute. 
+
+For example, `link` command supports `prelink` and `postlink` hooks to run before and after linking is done.
+
+These are the only ones supported by CLI at the moment. Depending on the packages used in your project, you may find other hooks supported to.
+
+> Note: This has nothing to do with React Hooks.
+
+## Migrating from `rnpm` configuration
+
+The changes are mostly cosmetic so the migration should be pretty straight-forward.
+
+### Changing the configuration
+
+Properties were renamed. Look at the following example for the differences.
+
+```json
+{
+  "rnpm": {
+    "ios": {},
+    "android": {},
+    "assets": ["./path-to-assets"],
+    "hooks": {
+      "prelink": "./path-to-a-postlink-hook"
+    }
+  }
+}
+```
+
+to `react-native.config.js`
+
+```js
+module.exports = {
+  dependency: {
+    platforms: {
+      ios: {},
+      android: {},
+    },
+    assets: ['./path-to-assets'],
+    hooks: {
+      prelink: './path-to-a-postlink-hook'
+    }
+  }
+};
+```
+
+### Asking for params while linking has been removed
+
+If your library needs it, do not upgrade over to the new config format.
```

Comment:
Do we provide any alternative?

Reply:
Yes and is listed below (custom solution). I don't think this was super useful and is actually going to be harder to maintain in the future.